src/directives/formioElement.js

```diff
@@ -11,9 +11,18 @@ module.exports = [
         element.replaceWith($compile($templateCache.get(scope.template))(scope));
         scope.$emit('formElementRender', element);
       },
-      controller: function() {
-        // This is required for some reason as it will occasionally throw an error without it.
-      }
+      controller: [
+        '$scope',
+        function(
+          $scope
+        ) {
+          $scope.$watchCollection('data.' + $scope.component.key, function(_new, _old) {
```

Comment:
I haven't experienced this yet, but I'm sure its possible with larger forms (my test form has 1 of each component)..

If the data collection is watched, then each change would require a total diff of the old and new object to determine what property was changed would it not? That is why I put a watch on each component of the data obj, rather than the data obj itself.

Thoughts?